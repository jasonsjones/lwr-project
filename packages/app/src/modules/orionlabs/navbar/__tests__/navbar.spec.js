import { createElement } from 'lwc';
import Navbar from 'orionlabs/navbar';

async function createComponent(prop = {}) {
    const element = createElement('orionlabs-navbar', {
        is: Navbar
    });

    Object.assign(element, prop);
    document.body.appendChild(element);
    await Promise.resolve();

    return element;
}

describe('NavBar component', () => {
    afterEach(() => {
        while (document.body.firstChild) {
            document.body.removeChild(document.body.firstChild);
        }
    });

    it('renders', async () => {
        const element = await createComponent();
        expect(element).toMatchSnapshot();
    });
});
