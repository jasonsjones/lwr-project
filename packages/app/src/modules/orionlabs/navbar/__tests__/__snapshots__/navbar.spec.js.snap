// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component renders 1`] = `
<orionlabs-navbar>
  #shadow-root(open)
    <header
      class="px-8 py-4 bg-primary-900 text-gray-200"
    >
      <nav
        class="flex gap-8"
      >
        <a
          href="/"
        >
          <div
            class="flex flex-col items-end"
          >
            <div
              class="text-3xl font-bold uppercase hover:text-white"
            >
              Orion
            </div>
            <div
              class="-mr-2 px-2 text-lg text-white font-semibold bg-primary-400 rounded-full"
            >
              labs
            </div>
          </div>
        </a>
        <div
          class="flex grow items-center justify-between"
        >
          <div
            class="flex gap-8"
          >
            <orionlabs-nav-link>
              #shadow-root(open)
                <a
                  class="text-xl pb-1 hover:text-white"
                >
                  About
                </a>
            </orionlabs-nav-link>
          </div>
          <div
            class="flex gap-8"
          >
            <orionlabs-nav-link>
              #shadow-root(open)
                <a
                  class="text-xl pb-1 hover:text-white"
                >
                  Sign Up
                </a>
            </orionlabs-nav-link>
            <orionlabs-nav-link>
              #shadow-root(open)
                <a
                  class="text-xl pb-1 hover:text-white"
                >
                  Log In
                </a>
            </orionlabs-nav-link>
          </div>
        </div>
      </nav>
    </header>
</orionlabs-navbar>
`;
